### ISSUE TITLE
Image icon in a message preview appears and disappears

### ISSUE THREAD

**Issue by sanmiguel2019** [tinode/webapp#70]
I see this problem only in my Android mobile phone but not in computer. In both devices I use Chrome browser.

I open the webapp and in the contacts list I only see the following message preview: "Hello how are you?"

![app5](https://user-images.githubusercontent.com/54183931/116422352-91570900-a837-11eb-9ab4-11ceee196013.png)

I tap on the topic and see that the last message contains not only "How are you?" but also an image. I return to the contacts list and now message preview is displayed correctly (with image).

![app6](https://user-images.githubusercontent.com/54183931/116422393-99af4400-a837-11eb-880b-feb547f296bf.png)

I close the webapp and when it open again, I see the same problem in the contacts list - only "How are you?", without image icon.

**Comment by or-else** [comment]
It's probably not a bug. It's most likely a problem with caching material icons font.

**Comment by sanmiguel2019** [comment]
Probably. Today I don't see this problem anymore.

**Comment by sanmiguel2019** [comment]
I only wonder, if it was a caching issue, why I didn't see the text "Imagen".

**Comment by or-else** [comment]
Do you still see the problem? Maybe you can debug it?

**Comment by sanmiguel2019** [comment]
Yes, I still see the problem. Now I see it not only on my computer too. I can't understand the pattern of this problem.

I reload the webapp. The last message in all 4 topics is an image but only 1 message preview shows it. I see this:

![app9](https://user-images.githubusercontent.com/54183931/116816975-2faadd80-ab5c-11eb-8da0-12dee14985a1.png)

When I click on each topic in the contacts list, image icon + word "Imagen" immediately appears 

![app10](https://user-images.githubusercontent.com/54183931/116817038-86181c00-ab5c-11eb-9ab9-738a87b868d0.png)


and stays there until I reload the webapp. After reloading the same problem - only 1 topic shows image in the preview,

**Comment by or-else** [comment]
I still cannot reproduce. When you see it please do the following:

Open Developer Tools in your browser, then use element picker to see the markup of the broken element. See if markup is actually missing:

<img width="587" alt="Screenshot 2021-05-02 at 09 19 46" src="https://user-images.githubusercontent.com/5938143/116820072-ca66e480-ab7b-11eb-8f01-c6920c17d8fa.png">

If markup is present, open the IndexedDB and find the last message for your chosen topic. See if the message is present and not broken.

<img width="740" alt="Screenshot 2021-05-02 at 09 24 34" src="https://user-images.githubusercontent.com/5938143/116820130-1023ad00-ab7c-11eb-901f-bbe1fba95a9c.png">

**Comment by sanmiguel2019** [comment]
The markup is not present:

![app11](https://user-images.githubusercontent.com/54183931/116820919-5aeaf800-ab6f-11eb-9aa8-df5dfbeafe41.png)

When I click on the topic, the markup immediately appears:

![app12](https://user-images.githubusercontent.com/54183931/116821501-fe3d0c80-ab71-11eb-8c5f-8a81841883a7.png)



I was a way for some hour and now when I reloaded the webapp, appeared only i markup on 4th topic. Before was 2nd and 4th.

Message:

![app13](https://user-images.githubusercontent.com/54183931/116821453-d51c7c00-ab71-11eb-8ebe-94b165c7a60b.png)

**Comment by sanmiguel2019** [comment]
I deleted the database and after reloading none of the topics has "picture" markup until I click on the topic.

**Comment by or-else** [comment]
> The markup is not present:

That's interesting. That means the message is missing but in some non-obvious way.

> I deleted the database and after reloading none of the topics has "picture" markup until I click on the topic.

That's normal. The messages have not been received by the client yet.

**Comment by or-else** [comment]
Here is what you can do. When the markup is missing don't click on the topic. Open IndexedDB and find the last message in the given topic. See if the message content is present and it's valid Drafty content.

Also check if you can reproduce the problem when the message content is plain text (not Drafty).

**Comment by sanmiguel2019** [comment]
Last message with image and text. Contacts list:

![app14](https://user-images.githubusercontent.com/54183931/116822527-31ce6580-ab77-11eb-8ebd-02e8eb95474e.png)

Message:

![app16](https://user-images.githubusercontent.com/54183931/116822533-43177200-ab77-11eb-8bcd-c3244910d76f.png)


After clicking on the topic:

![app15](https://user-images.githubusercontent.com/54183931/116822542-5296bb00-ab77-11eb-9332-b945536fb688.png)

**Comment by or-else** [comment]
The markup was completely missing before, now only the icon is missing. Can you **consistently** check for markup presence and then the message content in db? I.e. always the image alone, or always the image + text (more useful).

**Comment by sanmiguel2019** [comment]
No. In previous examples messages contained only images without text. That's why the markup was completely missing. In the last example message contains image and text "next test". So, if a message contains some text, it appears always even if message contains image. Only material icon and wird "Picture" is missing

This morning downloaded a fresh copy of Tinode's webapp to exclude a possibility that the problem may be caused because of some changes in my css file or something. But it's not the case. I saw the same problem.

OK, I will do more checks of markup presence.

**Comment by sanmiguel2019** [comment]
I hope you understand what I mean:

![app17](https://user-images.githubusercontent.com/54183931/116823234-3eed5380-ab7b-11eb-8791-67e009670e2e.png)

Thats why beside of translations I suggested to use different color of `formatMessage` in messages previews. if not in the screenshots that I added "Imagen" looks like the first word of the message text.

**Comment by or-else** [comment]
>  I saw the same problem.

I did not attempt to fix it. I cannot reproduce it and consequently cannot fix. You are able to reproduce it and I'm asking you to do things to let me find the cause. That's why I'm asking you to follow these steps:

* See if the markup is missing
* Without clicking the topic, get the message from the IndexedDB
* It would also be helpful to check if the message in IndexedDB changes after you click to topic.

Do it consistently on a message with image+text.

**Comment by or-else** [comment]
> I suggested to use different color 

Let's solve the problem first.

**Comment by sanmiguel2019** [comment]
I had no time to play with IndexedDB but I will do it today but before I do it I want to comment the following.

But I guess i have understand the pattern of the problem and why you can not repruduce it. The problem depends of the last messge in the topic id it's sent or received.

Example:

Alice sends 3 messages - to Bob, Dave and Carol. All the messages contains an image and in her contact list appears a corerct markup. Alice can reload the webapp as many times as she wants and this markup in the contact list will always appear.

However, the recients - Bob, Dave and Carol will see the markup only after clicking on the topic. And if they reload the webapp, the markup will disappear again.

If the last message with image or attachment is SENT (not RECEIVED), the markup will always be visible but if the last message is RECEIVED, there will be a problem.

And I noticed one more thing - the tick icon does not work corerctly in the contacts list.
Alice sends a message to Bob. Bob is active in the webapp. In Bob's contact list appears badge indicating that he has a new message from Alice. But in Alice's contact list single tick does not change to double (message received).

**Comment by or-else** [comment]
Thanks, I can reproduce it now.

**Comment by or-else** [comment]
The disappearing icon is fixed in https://github.com/tinode/tinode-js/commit/3522c0f9ee7105c13484d33fbad83914ecd99918

**Comment by or-else** [comment]
> But in Alice's contact list single tick does not change to double (message received)

That's not a big per se. It's missing functionality. The double check indicates that the message was received and saved by the recipient's device. In this case the message was not received by the device, just the notification was. Hence the single check is correct.

**Comment by sanmiguel2019** [comment]
> The disappearing icon is fixed in [tinode/tinode-js@3522c0f](https://github.com/tinode/tinode-js/commit/3522c0f9ee7105c13484d33fbad83914ecd99918)

I don´t know how to test it. Tinode SDK is not updated yet. I downloaded drafty.js file and replaced this file in /node_modules/tinode-sdk/src/drafty.js but after rebuilding the app I see the same problem.

**Comment by or-else** [comment]
Everything is updated.

**Comment by sanmiguel2019** [comment]
I am not sure if I have to open a new issue but I still see the same problems in the contacts list. Messages preview and new messages indicadors disappear after reloading the webapp. Downloaded the last version of Tinode's Webapp. The same problems I see on sandbox.tinode.co

Bob sends message to Alice:

![app20](https://user-images.githubusercontent.com/54183931/118693544-0dbf8500-b803-11eb-95f2-4fe0ca9ae35b.png)


Alice does not read the new message but reloads the webapp: https://sandbox.tinode.co/#contacts

After reloading there is no new messages from Bob anymore, and there is no message preview under Bob's name either.

![app22](https://user-images.githubusercontent.com/54183931/118694898-8246f380-b804-11eb-95fa-cbf0c14c274e.png)


Bob sends one more message to Alice. Now Alice sees that she has 2 new messages from Bob but there is still no message preview.

![app21](https://user-images.githubusercontent.com/54183931/118694743-4dd33780-b804-11eb-9813-8b31da578d57.png)


Alice clicks on the Bob's topic. Immediately appears message preview and badge showing that Bog has 2 new messages from Alice.

Database does not update.

I see the same read and seq numbers before and after after receiving messages and after reloading the webapp:

![app23](https://user-images.githubusercontent.com/54183931/118695647-55dfa700-b805-11eb-90bb-4001bcef1073.png)


Only when Alice clicks on the topic, these numbers change.

![app24](https://user-images.githubusercontent.com/54183931/118695704-62fc9600-b805-11eb-843b-aa53baa0987a.png)

**Comment by or-else** [comment]
Please open a new issue if it is a new issue. Opening issues is free and simple.
